dev-docs/design/how-lit-works.md

```diff
@@ -0,0 +1,320 @@
+
+### 2.2. Create a `<template>` element
+
+The HTML string from step 2.1 is used to set the `innerHTML` of a new `<template>` element. This causes the browser to parse the template HTML. Text-position markers that were written as comments are parsed into comment nodes.
+
+Because this step does not use any provided values (and because templates are inert), this step is safe from XSS vulnerabilities or other malicious input.
+
+### 2.3. Create a lit-html `Template`
+
+The `Template` class does most of the heavy-lifting for preparing a template. It walks the tree of nodes in a `<template>` element finding the markers inserted in step 2.1. If a marker is found, either in an attribute, as a marker comment node, or in text content, the depth-first index of the node is recorded, along with the type of expression (`'text'` or `'attribute'`) and the name of the attribute. The metadata containing the depth-first index of the node and type of expression is called a `TemplatePart` in the source.
+
+When traversing the tree of nodes, `Template` does different work for specific Node types:
+
+#### Element
+
+For each attribute on the element, if it has a `$lit$` suffix, then the expression represents either a `PropertyPart`, `BooleanAttributePart`, `EventPart`, or `AttributePart`. These can be distinguished by the first character in the attribute name ([see authored colum in Parts table](#parts)).
+
+Bound attributes are removed and a `TemplatePart` is created that records its location and name.
+
+An attribute on the element may also represent an `ElementPart` in the case where the attribute name starts with the marker sentinel value. E.g. `<div ${} ${}>` is marked up as `<div lit$random$0="" lit$random$1="">`.
+
+#### Text
+
+Usually text-position markers will be comment nodes, but inside `<style>`, `<script>`, `<textarea>`, `<title>` tags the markup that _looks_ like a comment (`<!-- lit -->`) will just be inserted as _text_ in the script or style. So we must search for the marker as text. If found we split the Text node and insert a comment marker.
+
+#### Comment
+
+A Comment is usually either an expression marker, or a user-written comment with no expression associated with them. Occasionally though a user may have written an expression inside a comment. This is especially easy to do with IDEs that help comment out code sections and are inline-html aware. We might see a comment like `<!--<div>${text}</div>-->` in the template text, so we must scan comments for marker text.
+
+A comment that matches an expression marker ([added in 2.1](#21-join-the-templates-string-literals-with-a-marker-string)), marks a `TemplatePart` in ChildPart position.
```

Comment:
Good catch. Also I need to rephrase the `marks a TemplatePart in ChildPart position` because really the TemplatePart **is** the metadata stored that later becomes a ChildPart.

Reply:
yup i definitely felt that throughout this whole section which i mention in the next comment regarding making it clear that `TemplatePart` is the template that is eventually used to make these different Parts.

Reply:
Done!